### PR TITLE
Fallback to the /ToUnicode map for TrueType fonts with (3, 1) and (1, 0) cmap-tables (issue 13316)

### LINKED ISSUE #13316: All of the Chinese character cannot be rendered
Attach (recommended) or Link to PDF file here:
[chinese.pdf](https://github.com/mozilla/pdf.js/files/6404389/download.1.pdf)

Configuration:
- Web browser and its version:90.0.4430.93
- Operating system and its version: mac 10.15.5
- PDF.js version: 2.9.146
- Is a browser extension:no

Steps to reproduce the problem:
1. The Chinese in the attachment cannot be rendered

Brower ✅
image

PDF.js ❌ [https://mozilla.github.io/pdf.js/web/viewer.html](https://mozilla.github.io/pdf.js/web/viewer.html)
image

What is the expected behavior? (add screenshot)
**The Chinese character can be rendered**
What went wrong? (add screenshot)
**All of the Chinese character cannot be rendered**
Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

### PR DESCRIPTION
**Pull request by Snuffleupagus**
In the PDF document some of the glyphs have bogus `differences`-entries[1] that cannot be resolved to valid glyph names, thus causing the glyph mapping to fail.
My initial idea was to use a similar approach as in the `PartialEvaluator._simpleFontToUnicode`-method, to extract the charCodes from those entries, however it turned out that that didn't actually help in this case (the mapping was still wrong).

To fix this I'm thus proposing that we fallback to the /ToUnicode map when no other useable data exists (e.g. no post-table), since it *hopefully* shouldn't make things any worse than leaving parts of the glyph map empty (which currently happens).

Fixes #13316

---
[1] As can be seem below, some of the entries are completely normal while others are non-standard:
```
Differences (array)
    0 = 65
    1 = /g5167
    2 = /space
    3 = /g11927
    4 = /g17737
    5 = /g11540
    6 = /g2180
    7 = /K
    8 = /P
    9 = /two
    10 = /zero
    11 = /one
    12 = /five
    13 = /four
    14 = /g6932
    15 = /g7246
    16 = /g1691
    17 = /g2343
    18 = /g14792
    19 = /g3325
    20 = /g4280
    21 = /g20383
    22 = /g18166
    23 = /g16988
    24 = /g17943
    25 = /g19223
    26 = /g10830
    27 = 97
    28 = /g982
    29 = /g1226
    30 = /g5059
    31 = /g2677
    32 = /g1042
    33 = /g11568
    34 = /L
    35 = /three
    36 = /seven
    37 = /g2364
    38 = /g12063
    39 = /g5356
    40 = /g2173
    41 = /g17877
    42 = /g7273
    43 = /g7647
    44 = /g7224
    45 = /g19327
    46 = /g5054
    47 = /g2342
    48 = /g10136
    49 = /g6856
    50 = /g13381
    51 = /g7257
    52 = /g12093
    53 = /g2359
```

### CODE REVIEW

**Comment by Snuffleupagus** [comment]
@brendandahl This patch does feel really hacky and somewhat arbitrary, so please just tell me if this patch looks stupid :-)

File: src/core/fonts.js [hunk 1]
```diff
@@ -2651,6 +2651,25 @@ class Font {
...
+            // Not a valid glyph name, fallback to using the /ToUnicode map
+            // when no post-table exists (fixes issue13316_reduced.pdf).
+            if (
+              !properties.glyphNames &&
+              properties.hasIncludedToUnicodeMap &&
+              !(this.toUnicode instanceof IdentityToUnicodeMap)
+            ) {
+              const unicode = this.toUnicode.get(charCode);
+              if (unicode) {
+                unicodeOrCharCode = unicode.charCodeAt(0);
```
Comment (brendandahl) [inline src/core/fonts.js:2664]: Do we want codePointAt here instead?

**Review by brendandahl — APPROVED** [review]
This looks reasonable, pdfium has a [similar path](https://pdfium.googlesource.com/pdfium/+/5fa76bf3acca3391fa4a99ad1f3e85063249b252/core/fpdfapi/font/cpdf_truetypefont.cpp#143).

File: src/core/fonts.js (hunk 1 above)
Comment (Snuffleupagus) [reply src/core/fonts.js]: Yes, that seems totally reasonable.
Fixed now; and that thanks for the review!

**Pull request merged**